qtrees/data_processor.py

```diff
@@ -0,0 +1,311 @@
+import pandas as pd
+import numpy as np
+from functools import reduce
+import datetime as dt
+from typing import Union, Optional
+from sklearn.preprocessing import OrdinalEncoder
+
+ROLLING_WINDOW = 7
+FORECAST_HORIZON = 14
+AUTOREGRESSIVE_LAG = 3
+
+class Data_loader:
```

Comment:
Insgesamt (auch wenn zugegeben im Repo nicht ganz durchgezogen, aber als Reviewer darf ich das kritisieren :)):
- je Klasse und Methode gern ein Docstring. Im Minimum eine kurze Beschreibung. Ideal (vor allem bei wichtigen Methoden die von außerhalb der Datei verwendet werden) auch mit Beschreibung Input, Returns. Ich nutze da gern die Numpy convention (siehe nächster Kommentar)
-

Reply:
"""
    Brief description of the function.

    Longer and more detailed description of the function.

    Parameters
    ----------
    param1 : type
        Description of param1.
    param2 : type, optional
        Description of param2. Defaults to None.

    Returns
    -------
    type
        Description of the return value.

    Raises
    ------
    ExceptionType
        Why and when this exception is raised.

    Examples
    --------
    Example usage of the function.
    """